storage/dataflux/sequential.go

```diff
@@ -24,42 +24,44 @@ import (
 )
 
 const (
-	// defaultPageSize specifies the number of object results to include on a single page.
-	defaultPageSize = 5000
+	// seqDefaultPageSize specifies the number of object results to include on a single page for sequential listing.
+	seqDefaultPageSize = 5000
 )
 
 // sequentialListing performs a sequential listing on the given bucket.
 // It returns a list of objects and the next token to use to continue listing.
 // If the next token is empty, then listing is complete.
 func (c *Lister) sequentialListing(ctx context.Context) ([]*storage.ObjectAttrs, string, error) {
 	var result []*storage.ObjectAttrs
-	var objectsListed int
+	var objectsIterated int
 	var lastToken string
 	objectIterator := c.bucket.Objects(ctx, &c.query)
 	objectIterator.PageInfo().Token = c.pageToken
-	objectIterator.PageInfo().MaxSize = defaultPageSize
+	objectIterator.PageInfo().MaxSize = seqDefaultPageSize
 
 	for {
```

Comment:
We want the list to be a multiple of number_of_objects_gcs_api_fetches  in single call to make sure all the objects for a given token is listed.  Also when we set MaxSize to 5000, when request is asking for ACLs only 2000 objects are returned. 
The reason we are making doSeqListing return just one page is that the api call can fetch either 2000 or 5000 objects and still fetch all objects linked to the given token.
Updated the comment. 

Let me know what you think here.

Reply:
> sending in batchSize as a minSize to the doSeqListing and stopping there

By sending in a minimum size we can check here 
```
   if objectIterator.PageInfo().Remaining() == 0 { // check if we have reached at least the amount of objects we want here
			...
	}
```
I don't think we have public documentation on the page size, and it seems like an internal detail we shouldn't rely on, in fact, I don't think GCS guarantees anywhere publicly that each page will be of equal size. Please correct me if you know otherwise. This code doesn't seem to be relying on that as is, so it's okay to leave as is, but can we more specifically say in the comment that `listNextPageSequentially` will make at most one network call to GCS and will exhaust all objects currently held in the iterator?

In this vein, can we also add to the docs for `ListerInput.SkipDirectoryObjects` that directory objects count in the `batchSize`? Otherwise users may assume that we are filtering them server-side. We probably should mention it in `ListerInput.BatchSize` as well.

Also, in `ListerInput.BatchSize`, I see it mentions `The number of objects returned will be rounded up to a multiple of gcs page size.`"gcs page size" could be confusing to users, especially since as mentioned above, there is no exact value for this concept. Instead, can we rephrase it, maybe:

```
	// BatchSize is the minimum number of objects to list in each batch. 
	// The number of objects returned in a batch will be rounded up to 
        // include all the objects received in the last request to GCS.
	// By default, the Lister returns all objects in one batch. 
	// Optional.
```

Reply:
Done.